pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:dartx/dartx.dart';
+import 'package:pokedex_async_redux/api/model/details.dart';
+import 'package:pokedex_async_redux/features/pokemon_details/widget/pokemon_ability.dart';
+import 'package:pokedex_async_redux/features/pokemon_details/widget/pokemon_general.dart';
+import 'package:pokedex_async_redux/features/pokemon_details/widget/pokemon_moves.dart';
+import 'package:pokedex_async_redux/features/pokemon_details/widget/pokemon_stats.dart';
+import 'package:pokedex_async_redux/utilities/async.dart';
+import 'package:pokedex_async_redux/utilities/colors.dart';
+import 'package:pokedex_async_redux/utilities/constants.dart';
+import 'package:flutter_icons_null_safety/flutter_icons_null_safety.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:flutter/material.dart';
+
+class PokemonDetailsPage extends StatelessWidget {
+  const PokemonDetailsPage({
+    required this.pokemonDetails,
+    required this.pokeName,
+    required this.pokemonImage,
+    Key? key,
+  }) : super(key: key);
+
+  final Async<PokemonDetails> pokemonDetails;
+  final String pokeName;
+  final String pokemonImage;
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text(appBarTitle),
+        backgroundColor: appDefaultColor,
+      ),
+      body: pokemonDetails.when(
+        loading: () => const Center(child: CircularProgressIndicator(color: appDefaultColor)),
+        error: (errorMessage) {
+          WidgetsBinding.instance
+              .addPostFrameCallback((_) => _showErrorMessageSnackbar(context, errorMessage ?? emptyString));
+          return const Center(child: Text(noPokemonsAvailable));
+        },
+        (data) => SingleChildScrollView(
+          child: Center(
+            child: Column(
+              children: [
+                const SizedBox(height: 30),
+                Center(
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      const Icon(MaterialCommunityIcons.pokemon_go, color: appDefaultColor),
+                      const SizedBox(width: 8),
```

Comment:
kindly make its value double since it's basically asking for a double value 🙂 

same goes to other value assignment you made for this pr 🙂